### PR TITLE
Optimize batchnorm

### PR DESCRIPTION
**Pull request by primenumber**
## Motivation and Context

This PR optimize batch normalization operator.
Optimized operator calculate just multiply and add, and quit to use additional temporary buffer.

## Description

- Introduce `BatchNormalizationOptimized` operator
- convert `BatchNormalization` operator into optimized one if it is not merged into previous convolution
- Update `func_BatchNormalization`
- Remove temporary buffer for `func_BatchNormalization`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: dlk/python/dlk/templates/src/func/generic/batch_normalization.cpp [hunk 1]
```diff
@@ -20,32 +20,20 @@ limitations under the License.
 #include "func/batch_normalization.h"
 #include "time_measurement.h"
 
-static const auto scale = std::make_unique<float[]>(MAX_IN_C);
-static const auto shift = std::make_unique<float[]>(MAX_IN_C);
-
 void func_BatchNormalization(const TensorView<T_FLOAT, MemoryLayout::NHWC>& input,
```
Comment (tkng) [inline dlk/python/dlk/templates/src/func/generic/batch_normalization.cpp]: For consistency, it might be better to rename this function to `func_BatchNormalizationOptimized`. What do you think?

Comment (primenumber) [reply dlk/python/dlk/templates/src/func/generic/batch_normalization.cpp]: Renamed.

Comment (tkng) [reply dlk/python/dlk/templates/src/func/generic/batch_normalization.cpp]: thanks!
